StoreApp/Controllers/AccountController.cs

```diff
@@ -140,20 +140,22 @@ public ActionResult UserOrders(int? id)
         {
             var orders = webMapper.config.Map<IEnumerable<OrderDTO>, IEnumerable<OrderViewModel>>(orderService.GetAll().Where(o => id == null ? o.UserId != id : o.UserId == id));
 
-            foreach (var order in orders)
-            {
-                order.OrderDetails = webMapper.config.Map<IEnumerable<OrderDetailDTO>, IEnumerable<OrderDetailsViewModel>>(orderDetailService.GetAll().Where(o => o.OrderId == order.Id));
-
-                foreach (var details in order.OrderDetails)
-                {
-                    details.Product = webMapper.config.Map<ProductDTO, ProductViewModel>(productService.Get(details.ProductId));
-                    details.Product.Unit = unitService.Get(details.Product.UnitId).Name;
-                }
-            }
+            orders.ToList().ForEach(o => GetOrderDatails(o));
 
             return View(orders);
         }
 
+        private void GetOrderDatails(OrderViewModel order)
+        {
+            order.OrderDetails = webMapper.config.Map<IEnumerable<OrderDetailDTO>, IEnumerable<OrderDetailsViewModel>>(orderDetailService.GetAll().Where(o => o.OrderId == order.Id));
```

Comment:
Try to avoid using long strings. It's difficult to read them

Reply:
fixed